File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnResponseMeta.java

package com.gracenote.gnsdk;

public class GnResponseMeta extends GnDataObject {
  private transient long swigCPtr;

  protected GnResponseMeta(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnResponseMeta_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnResponseMeta obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnResponseMeta(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public static String gnType() {
    return gnsdk_javaJNI.GnResponseMeta_gnType();
  }

  public static GnResponseMeta from(GnDataObject obj) throws com.gracenote.gnsdk.GnException {
    return new GnResponseMeta(gnsdk_javaJNI.GnResponseMeta_from(GnDataObject.getCPtr(obj), obj), true);
  }

  public GnResponseMeta() {
    this(gnsdk_javaJNI.new_GnResponseMeta(), true);
  }

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnResponseMeta.cs

namespace GracenoteSDK {

public class GnResponseMeta : GnDataObject {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnResponseMeta(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnResponseMeta_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnResponseMeta obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnResponseMeta() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnResponseMeta(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public static string GnType() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnResponseMeta_GnType();
    return ret;
  }

  public static GnResponseMeta From(GnDataObject obj) {
    GnResponseMeta ret = new GnResponseMeta(gnsdk_csharp_marshalPINVOKE.GnResponseMeta_From(GnDataObject.getCPtr(obj)), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

  public GnResponseMeta() : this(gnsdk_csharp_marshalPINVOKE.new_GnResponseMeta(), true) {
  }

}

}
